runtime/src/master.rs

```diff
@@ -0,0 +1,695 @@
+{
+    type Event: From<Event<Self>> + Into<<Self as system::Trait>::Event>;
+
+    /// The dispatchable that master may call as Root. It is possible to use another type here, but
+    /// it's expectected that your runtime::Call will be used.
+    type Call: Parameter + Dispatchable<Origin = Self::Origin> + GetDispatchInfo;
+}
+
+decl_storage! {
+    trait Store for Module<T: Trait> as Master {
+        pub Members config(members): Membership;
+        pub Round: u64;
+    }
+}
+
+decl_error! {
+    pub enum MasterError for Module<T: Trait> {
+        /// The account used to submit this vote is not a member of Master.
+        NotMember,
+        /// This proposal does not yet have enough votes to be executed.
+        InsufficientVotes,
+        /// One of the signatures provided is invalid.
+        /// Hint: Is everyone voting for the current round?
+        BadSig,
+    }
+}
+
+decl_event! {
+    pub enum Event<T>
+    where
+        <T as Trait>::Call
+    {
+        /// A proposal succeeded and was executed. The dids listed are the members whose votes were
+        /// used as proof of authorization. The hash of the proposal is provided.
+        Executed(Vec<Did>, Box<Call>),
+        /// The membership of Master has changed.
+        UnderNewOwnership,
+    }
+}
+
+decl_module! {
+    pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+        type Error = MasterError<T>;
+
+        fn deposit_event() = default;
+
+        /// Execute a proposal that has received enough votes. The proposal is a serialized Call.
+        /// This function can be called by anyone, even someone who is not a member of Master.
+        ///
+        /// After a sucessful execution, the current round of votes is cleared and round number is
+        /// increased.
+        #[weight = 10_000 + proposal.get_dispatch_info().weight + 1_000 * auth.len() as u64]
+        pub fn execute(
+            origin,
+            proposal: Box<<T as Trait>::Call>,
+            auth: PMAuth,
+        ) -> DispatchResult {
+            Module::<T>::execute_(origin, proposal, auth)
+        }
+
+        /// Root-only. Sets the members and vote requirement for master. Increases the round number
+        /// and removes the votes for the prevous round.
+        ///
+        /// ```
+        /// # use dock_testnet_runtime::master::Membership;
+        /// # extern crate alloc;
+        /// # use alloc::collections::BTreeSet;
+        /// #
+        /// // Setting the following membership will effectively dissolve the master account.
+        /// Membership {
+        ///     members: BTreeSet::new(),
+        ///     vote_requirement: 1,
+        /// };
+        /// ```
+        ///
+        /// Setting the vote requirement to zero grants unrestricted root access to
+        /// all accounts. It is not recomended to set the vote requirement to zero on a
+        /// production chain.
+        #[weight = 10_000 + 1_000 * membership.members.len() as u64]
```

Comment:
You have 1 read and 2 writes here. This should set weight using `T::DbWeight::get().reads_writes(1, 2 + <some weight depending on the count of members, since members are usually 3-5 you can omit the offset>)`.

Reply:
done c159fed8

Reply:
Side note. My understanding is that merkle-trie root computation is the primary contributor to the cost of db-writes so byte size of the value doesn't matter as much. That is to say that if a weight was to be added for member set size, `T::DbWeight::get().reads_writes(1, 2) + weight` would be more fitting.

Reply:
Correct. And even i don't care much about the size of the value being written as these calls are privileged anyway. That's why I said you can omit the offset.